setup.py

```diff
@@ -1,25 +1,83 @@
-#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# setup.py
+#
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+AWS SAM Serverless Application Model
+"""
+import io
+import os
+import re
 
 from setuptools import setup, find_packages
 
 
-with open('README.md') as f:
-    readme = f.read()
+def read(*filenames, **kwargs):
```

Comment:
What advantage does this extra boilerplate give?

Reply:
this `read` better implement read files in different platforms that implement line separators differently.

> os.linesep
The string used to separate (or, rather, terminate) lines on the current platform. This may be a single character, such as '\n' for POSIX, or multiple characters, for example, '\r\n' for Windows. Do not use os.linesep as a line terminator when writing files opened in text mode (the default); use a single '\n' instead, on all platforms.
https://docs.python.org/2/library/os.html